### PR TITLE
fix(infiniteGrid): deal with defence code

### PR DESCRIPTION
**Pull request by sculove**
## Issue
#249

## Details
To restore status, I fix some code at onScroll handler

## Preferred reviewers
somebody
@happyhj @kishu

### CODE REVIEW

**Comment by kishu** [comment]
LGTM

File: test/unit/js/infiniteGrid.test.js [hunk 1]
```diff
@@ -444,15 +444,15 @@ test("restore status", function(assert) {
...
-			    var a =v.split(":");
-			    var val = $.trim(a[1]);
-			    if(!$.isEmptyObject(val)) {
-			        ht[a[0]] = $.trim(a[1]);
-			    }
+		    var a =v.split(":");
+		    var val = $.trim(a[1]);
+		    if(!$.isEmptyObject(val)) {
+		        ht[a[0]] = $.trim(a[1]);
+		    }
```
Comment (jongmoon) [inline test/unit/js/infiniteGrid.test.js]: Indentations of this modification are somewhat weird~

File: src/infiniteGrid.js [hunk 2]
```diff
@@ -335,8 +336,9 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document, "Outlayer"], function
 					});
 				}
 			} else {
-				if (this.isRecycling() && this._removedContent > 0 &&
-					(ele = this._topElement || this.getTopElement())) {
+				if (this.isRecycling() && this._removedContent > 0) {
+					ele = $.isEmptyObject(this._topElement) ?
+						this.getTopElement() : this._topElement;
```
Comment (jongmoon) [inline src/infiniteGrid.js]: 1. Is there case that `isBottomElement` or `isTopElement`  is empty object `{ }`?

2. Is there test cases related with it?

File: test/unit/js/infiniteGrid.test.js (hunk 1 above)
Comment (mixed) [inline test/unit/js/infiniteGrid.test.js]: @jongmoon 
I add issue relate to it.
https://github.com/naver/egjs/issues/256

Comment (jongmoon) [inline test/unit/js/infiniteGrid.test.js]: @mixed 👍

File: src/infiniteGrid.js (hunk 2 above)
Comment (sculove) [inline src/infiniteGrid.js]: isBottomElement and isTopElement are HTMLElement reference.

If page is leaved, HTMLElement is removed. 
When you come back page, you couldn't get previous HTMLElement reference.

and I fixed demo page and add some testcase in egjs-tc
I'm afraid that egjs-tc is private

File: test/unit/js/infiniteGrid.test.js (hunk 1 above)
Comment (sculove) [inline test/unit/js/infiniteGrid.test.js]: I am nice to look at my editor. it's depend on environments.

Comment (sculove) [inline test/unit/js/infiniteGrid.test.js]: @mixed @jongmoon good job

File: src/infiniteGrid.js (hunk 2 above)
Comment (jongmoon) [inline src/infiniteGrid.js]: @sculove Thank you for your kind explanation.

And I suggest... How about to cache `this._bottomElement` or `this._top_Element` if they are empty?

for example
```js
if ($.isEmptyObject(this._bottomElement)) {
  this._bottomElement = this.getBottomElement();
}
ele = this._bottomElement;
```

Comment (sculove) [inline src/infiniteGrid.js]: ok

File: src/infiniteGrid.js [hunk 3]
```diff
@@ -337,8 +339,10 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document, "Outlayer"], function
 				}
 			} else {
 				if (this.isRecycling() && this._removedContent > 0) {
-					ele = $.isEmptyObject(this._topElement) ?
-						this.getTopElement() : this._topElement;
+					if ($.isEmptyObject(this._topElement)) {
+						this._topElement = this.getBottomElement();
+					}
```
Comment (jongmoon) [inline src/infiniteGrid.js]: @sculove Fix above code!

```js
this._topElement = this.getTopElement();
```

**Comment by jongmoon** [comment]
LGTM

**Pull request merged**